src/unity/python/turicreate/toolkits/recommender/util.py

```diff
@@ -1260,10 +1260,16 @@ def check_type(arg, arg_name, required_type, allowed_types):
                 assert new_observation_data.num_rows() != 0
                 original_user_type = user_column.dtype
                 users[user_id] = user_column.astype(str)
+                user_type=str
 
             elif user_column.dtype != user_type:
                 users[user_id] = user_column.astype(user_type)
 
+        # Cast user specified in exclude to the appropriate type if necessary.
+        if user_id in exclude.column_names():
+            if exclude[user_id].dtype!=user_type:
```

Comment:
Usually, it's good practice to write a conjuncted conditional instead of writing a nested conditional. Replace `if A: if B: do stuff` with `if A and B: do stuff`. Note that Python has short-circuiting so if A is false, B will never get evaluated, like you intend here.

Reply:
Got it.